### PR TITLE
Add Archive delete

### PR DESCRIPTION
**Pull request by Raghav888**
1.  User can archive note
2. User can trash note
3. User can trash the archive note

### CODE REVIEW

File: src/context/note-context.js [hunk 1]
```diff
@@ -11,26 +11,51 @@ const { createContext, useContext, useEffect, useReducer } = require("react");
 const NoteContext = createContext();
 
 const NoteContextProvider = ({ children }) => {
-  const api =
+  const api_note =
     "https://my-json-server.typicode.com/Raghav888/mynoteappAPI/notes";
+  const api_archive =
+    "https://my-json-server.typicode.com/Raghav888/mynoteappAPI/archive";
+  const api_trash =
+    "https://my-json-server.typicode.com/Raghav888/mynoteappAPI/trash";
```
Comment (sushil-mod) [inline src/context/note-context.js:19]: I liked you are using your own json

File: src/context/note-context.js [hunk 2]
```diff
@@ -11,26 +11,51 @@ const { createContext, useContext, useEffect, useReducer } = require("react");
...
     tags: [],
+    trashData: [],
+    archiveData: [],
   });
 
   useEffect(() => {
     (async () => {
       try {
-        const response = await axios.get(api);
+        const response = await axios.get(api_note);
```
Comment (sushil-mod) [inline src/context/note-context.js:32]: this correct , you can also destructure ditrectly to data

**Review by sushil-mod — APPROVED** [review]
Note delete from archive working properly . just a suggestion . overall code is good

File: src/components/TrashData/TrashData.js [hunk 3]
```diff
@@ -1,38 +1,22 @@
...
-        trashdispatch({
-          type: "ADD_TRASH_DATA",
-          payload: { value: response.data },
-        });
-      } catch (error) {
-        console.log(error);
-      }
-    })();
-  }, []);
+  const { noteListState } = useNote();
```
Comment (munjalSaksham71) [inline src/components/TrashData/TrashData.js:8]: This is good. Just a suggestion for future use :

you can replace ```{ noteListState } ``` ```{noteListState: { trashData } }``` so for mapping you have  to use only ```trashData```

**Review by munjalSaksham71 — APPROVED** [review]
Functionality works absolutely fine. Also code is undersatndable. good to merge

**Pull request merged**